From b40b5b92a571f9a0e9811cbd6abb2a19017cc7f8 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Fri, 26 Aug 2011 14:23:53 +0800
Subject: [PATCH 2/2] powerpc/p2020: update p2020 DTS for PM support

Update the p2020 DTS for PM support. Currently we support:
- sleep
- hibernation
- wake up on mpic timer
- wake up on magic packet

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/p2020ds_36b.dts  |   16 ++++++++++++++++
 arch/powerpc/boot/dts/p2020rdb.dts     |   28 ++++++++++++++++++++++++++++
 arch/powerpc/boot/dts/p2020rdb_36b.dts |   28 ++++++++++++++++++++++++++++
 3 files changed, 72 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/boot/dts/p2020ds_36b.dts b/arch/powerpc/boot/dts/p2020ds_36b.dts
index bbfc159..dc49839 100644
--- a/arch/powerpc/boot/dts/p2020ds_36b.dts
+++ b/arch/powerpc/boot/dts/p2020ds_36b.dts
@@ -412,6 +412,8 @@
 			phy-handle = <&phy0>;
 			ptimer-handle = < &ptp_timer >;
 			phy-connection-type = "rgmii-id";
+			fsl,magic-packet;
+			clk-handle = <&etsec1_clk>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -457,6 +459,8 @@
 			phy-handle = <&phy1>;
 			ptimer-handle = < &ptp_timer >;
 			phy-connection-type = "rgmii-id";
+			fsl,magic-packet;
+			clk-handle = <&etsec2_clk>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -487,6 +491,8 @@
 			phy-handle = <&phy2>;
 			ptimer-handle = < &ptp_timer >;
 			phy-connection-type = "rgmii-id";
+			fsl,magic-packet;
+			clk-handle = <&etsec3_clk>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -536,6 +542,16 @@
 		power@e0070{
 			compatible = "fsl,mpc8548-pmc";
 			reg = <0xe0070 0x20>;
+
+			etsec1_clk: soc-clk@24{
+				fsl,pmcdr-mask = <0x00000080>;
+			};
+			etsec2_clk: soc-clk@25{
+				fsl,pmcdr-mask = <0x00000040>;
+			};
+			etsec3_clk: soc-clk@26{
+				fsl,pmcdr-mask = <0x00000020>;
+			};
 		};
 
 		timer@41100 {
diff --git a/arch/powerpc/boot/dts/p2020rdb.dts b/arch/powerpc/boot/dts/p2020rdb.dts
index 6007ef4..0b6fb7d 100644
--- a/arch/powerpc/boot/dts/p2020rdb.dts
+++ b/arch/powerpc/boot/dts/p2020rdb.dts
@@ -429,6 +429,8 @@
 			fixed-link = <1 1 1000 0 0>;
 			phy-connection-type = "rgmii-id";
 			ptimer-handle = <&ptp_timer>;
+			fsl,magic-packet;
+			clk-handle = <&etsec1_clk>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -465,6 +467,8 @@
 			phy-handle = <&phy0>;
 			phy-connection-type = "sgmii";
 			ptimer-handle = <&ptp_timer>;
+			fsl,magic-packet;
+			clk-handle = <&etsec2_clk>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -494,6 +498,8 @@
 			phy-handle = <&phy1>;
 			phy-connection-type = "rgmii-id";
 			ptimer-handle = <&ptp_timer>;
+			fsl,magic-packet;
+			clk-handle = <&etsec3_clk>;
 		};
 
 		sdhci@2e000 {
@@ -521,6 +527,13 @@
 			fsl,channel-remap = <0x3>;
 		};
 
+		timer@41100 {
+			compatible = "fsl,mpic-global-timer";
+			reg = <0x41100 0x204>;
+			interrupts = <0xf7 0x2>;
+			interrupt-parent = <&mpic>;
+		};
+
 		mpic: pic@40000 {
 			interrupt-controller;
 			#address-cells = <0>;
@@ -552,6 +565,21 @@
 			interrupt-parent = <&mpic>;
 		};
 
+		power@e0070{
+			compatible = "fsl,p2020-pmc", "fsl,mpc8548-pmc";
+			reg = <0xe0070 0x20>;
+
+			etsec1_clk: soc-clk@24{
+				fsl,pmcdr-mask = <0x00000080>;
+			};
+			etsec2_clk: soc-clk@25{
+				fsl,pmcdr-mask = <0x00000040>;
+			};
+			etsec3_clk: soc-clk@26{
+				fsl,pmcdr-mask = <0x00000020>;
+			};
+		};
+
 		msi@41600 {
 			compatible = "fsl,p2020-msi", "fsl,mpic-msi";
 			reg = <0x41600 0x80>;
diff --git a/arch/powerpc/boot/dts/p2020rdb_36b.dts b/arch/powerpc/boot/dts/p2020rdb_36b.dts
index 741bb4e..9924718 100644
--- a/arch/powerpc/boot/dts/p2020rdb_36b.dts
+++ b/arch/powerpc/boot/dts/p2020rdb_36b.dts
@@ -417,6 +417,8 @@
 			fixed-link = <1 1 1000 0 0>;
 			phy-connection-type = "rgmii-id";
 			ptimer-handle = <&ptp_timer>;
+			fsl,magic-packet;
+			clk-handle = <&etsec1_clk>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -453,6 +455,8 @@
 			phy-handle = <&phy0>;
 			phy-connection-type = "sgmii";
 			ptimer-handle = <&ptp_timer>;
+			fsl,magic-packet;
+			clk-handle = <&etsec2_clk>;
 
 			mdio@520 {
 				#address-cells = <1>;
@@ -482,6 +486,8 @@
 			phy-handle = <&phy1>;
 			phy-connection-type = "rgmii-id";
 			ptimer-handle = <&ptp_timer>;
+			fsl,magic-packet;
+			clk-handle = <&etsec3_clk>;
 		};
 
 		sdhci@2e000 {
@@ -508,6 +514,13 @@
 			fsl,channel-remap = <0x3>;
 		};
 
+		timer@41100 {
+			compatible = "fsl,mpic-global-timer";
+			reg = <0x41100 0x204>;
+			interrupts = <0xf7 0x2>;
+			interrupt-parent = <&mpic>;
+		};
+
 		mpic: pic@40000 {
 			interrupt-controller;
 			#address-cells = <0>;
@@ -517,6 +530,21 @@
 			device_type = "open-pic";
 		};
 
+		power@e0070{
+			compatible = "fsl,p2020-pmc", "fsl,mpc8548-pmc";
+			reg = <0xe0070 0x20>;
+
+			etsec1_clk: soc-clk@24{
+				fsl,pmcdr-mask = <0x00000080>;
+			};
+			etsec2_clk: soc-clk@25{
+				fsl,pmcdr-mask = <0x00000040>;
+			};
+			etsec3_clk: soc-clk@26{
+				fsl,pmcdr-mask = <0x00000020>;
+			};
+		};
+
 		message@41400 {
 			compatible = "fsl,p2020-msg", "fsl,mpic-msg";
 			reg = <0x41400 0x200>;
-- 
1.7.0.4

